com.unity.ml-agents/Runtime/Sensor/RayPerceptionSensorComponentBase.cs

```diff
@@ -1,42 +1,112 @@
 using System;
 using System.Collections.Generic;
 using UnityEngine;
+using UnityEngine.Serialization;
 
 namespace MLAgents
 {
     public abstract class RayPerceptionSensorComponentBase : SensorComponent
     {
-        public string sensorName = "RayPerceptionSensor";
+        [HideInInspector]
+        [SerializeField]
+        [FormerlySerializedAs("sensorName")]
+        string m_SensorName = "RayPerceptionSensor";
+        internal string sensorName
+        {
+            get => m_SensorName;
+            set => m_SensorName = value;
+        }
 
+        //[HideInInspector]
```

Comment:
nit: remove

Reply:
Will do. I'm not quite clear why we have this in the first place for things like this:
https://github.com/Unity-Technologies/ml-agents/blob/9450d3fc0dda4547a14c5ed1b7e13fc6e3a15413/com.unity.ml-agents/Runtime/Policy/BehaviorParameters.cs#L35-L37
but then expose it in the editor: https://github.com/Unity-Technologies/ml-agents/blob/9450d3fc0dda4547a14c5ed1b7e13fc6e3a15413/com.unity.ml-agents/Editor/BehaviorParametersEditor.cs#L28

I had to remove it here to get the list to expand properly.